integration/query_compat_test.go

```diff
@@ -113,6 +119,216 @@ func TestQueryCompat(t *testing.T) {
 	t.Parallel()
 
 	testCases := map[string]queryCompatTestCase{
+		"ArrayOne": {
```

Comment:
I think we should move all test into the same file and function – it would make it harder to run only a subset of tests. Consider, for example, our existing `query_logical_compat_test.go` file. So, for example, all queries with arrays could be moved to `query_array_compat_test.go` file and `TestQueryArrayCompat` function

Reply:
Just noticed that myself as well. I created function for each operator like `TestQueryArrayCompatAll` `TestQueryArrayCompatSize`

Reply:
Having separate tests like `TestQueryArrayCompatAll` and `TestQueryArrayCompatSize` might be the reason why it slows down. I'll put them all in one func `TestQueryArrayCompat` as suggested.